File: backend/app/api/mentors.py
"""
导师信息API
支持动态获取导师信息、导师配置管理等功能
"""
import json
import logging
from typing import Dict, Any, List, Optional
from datetime import datetime

from fastapi import APIRouter, HTTPException, Query
from pydantic import BaseModel

from ..agents.agent_manager import agent_manager
from ..core.config import settings

logger = logging.getLogger(__name__)

# 创建路由
router = APIRouter(prefix="/mentors", tags=["mentors"])

# 数据模型
class MentorInfo(BaseModel):
    """导师信息模型"""
    agent_id: str
    name: str
    title: str
    description: str
    voice: str
    expertise: List[str]
    personality_traits: List[str]
    investment_style: str
    famous_quotes: List[str]
    color: str
    avatar: str
    enabled: bool
    priority: int
    registered_at: str

class MentorConfig(BaseModel):
    """导师配置模型"""
    agent_id: str
    name: str
    description: str
    priority: int
    enabled: bool
    voice: str

class MentorUpdateRequest(BaseModel):
    """导师更新请求模型"""
    name: Optional[str] = None
    description: Optional[str] = None
    priority: Optional[int] = None
    enabled: Optional[bool] = None
    voice: Optional[str] = None

class DynamicMentorRequest(BaseModel):
    """动态导师生成请求模型"""
    topic: str
    session_id: str

class DynamicMentorResponse(BaseModel):
    """动态导师生成响应模型"""
    mentors: List[MentorInfo]
    topic: str
    session_id: str
    generated_at: str

# 导师颜色映射
MENTOR_COLORS = {
    'buffett': '#3B82F6',      # 蓝色 - 价值投资
    'soros': '#10B981',        # 绿色 - 宏观投资
    'munger': '#8B5CF6',       # 紫色 - 多元思维
    'krugman': '#F59E0B',      # 橙色 - 宏观经济
}

# 导师头像映射
MENTOR_AVATARS = {
    'buffett': 'https://api.dicebear.com/7.x/adventurer/svg?seed=warren-buffett',
    'soros': 'https://api.dicebear.com/7.x/avataaars/svg?seed=george-soros',
    'munger': 'https://api.dicebear.com/7.x/big-ears/svg?seed=charlie-munger',
    'krugman': 'https://api.dicebear.com/7.x/croodles/svg?seed=paul-krugman',
}

def get_mentor_color(agent_id: str) -> str:
    """获取导师颜色"""
    if agent_id in MENTOR_COLORS:
        return MENTOR_COLORS[agent_id]
    # 为动态导师生成颜色
    colors = ['#3B82F6', '#10B981', '#8B5CF6', '#F59E0B', '#EF4444', '#8B5A2B']
    color_index = hash(agent_id) % len(colors)
    return colors[color_index]

def get_mentor_avatar(agent_id: str) -> str:
    """获取导师头像"""
    if agent_id in MENTOR_AVATARS:
        return MENTOR_AVATARS[agent_id]
    # 为动态导师生成头像
    avatar_styles = ['adventurer', 'avataaars', 'big-ears', 'croodles', 'fun-emoji', 'lorelei']
    style_index = hash(agent_id) % len(avatar_styles)
    return f'https://api.dicebear.com/7.x/{avatar_styles[style_index]}/svg?seed={agent_id}'

@router.get("/", response_model=List[MentorInfo])
async def get_all_mentors():
    """
    获取所有导师信息
    
    Returns:
        导师信息列表
    """
    try:
        logger.info("📋 获取所有导师信息")
        
        mentors = []
        for agent_id, agent in agent_manager.agents.items():
            agent_config = agent_manager.agent_configs.get(agent_id, {})
            agent_info = agent.get_agent_info()
            
            mentor_info = MentorInfo(
                agent_id=agent_id,
                name=agent_config.get('name', agent_info.get('name', agent_id)),
                title=agent_info.get('title', ''),
                description=agent_config.get('description', ''),
                voice=agent_config.get('voice', agent_info.get('voice', 'Cherry')),
                expertise=agent_info.get('expertise', []),
                personality_traits=agent_info.get('personality_traits', []),
                investment_style=agent_info.get('investment_style', ''),
                famous_quotes=agent_info.get('famous_quotes', []),
                color=get_mentor_color(agent_id),
                avatar=get_mentor_avatar(agent_id),
                enabled=agent_config.get('enabled', True),
                priority=agent_config.get('priority', 1),
                registered_at=agent_config.get('registered_at', datetime.now().isoformat())
            )
            mentors.append(mentor_info)
        
        # 按优先级排序
        mentors.sort(key=lambda x: x.priority)
        
        logger.info(f"✅ 成功获取 {len(mentors)} 个导师信息")
        return mentors
        
    except Exception as e:
        logger.error(f"❌ 获取导师信息失败: {e}")
        raise HTTPException(status_code=500, detail=f"获取导师信息失败: {str(e)}")

@router.get("/enabled", response_model=List[MentorInfo])
async def get_enabled_mentors():
    """
    获取所有启用的导师信息
    
    Returns:
        启用的导师信息列表
    """
    try:
        logger.info("📋 获取启用的导师信息")
        
        mentors = []
        enabled_agents = agent_manager.get_enabled_agents()
        
        for agent_id, agent in enabled_agents.items():
            agent_config = agent_manager.agent_configs.get(agent_id, {})
            agent_info = agent.get_agent_info()
            
            mentor_info = MentorInfo(
                agent_id=agent_id,
                name=agent_config.get('name', agent_info.get('name', agent_id)),
                title=agent_info.get('title', ''),
                description=agent_config.get('description', ''),
                voice=agent_config.get('voice', agent_info.get('voice', 'Cherry')),
                expertise=agent_info.get('expertise', []),
                personality_traits=agent_info.get('personality_traits', []),
                investment_style=agent_info.get('investment_style', ''),
                famous_quotes=agent_info.get('famous_quotes', []),
                color=MENTOR_COLORS.get(agent_id, '#6B7280'),
                avatar=MENTOR_AVATARS.get(agent_id, f'https://api.dicebear.com/7.x/adventurer/svg?seed={agent_id}'),
                enabled=True,
                priority=agent_config.get('priority', 1),
                registered_at=agent_config.get('registered_at', datetime.now().isoformat())
            )
            mentors.append(mentor_info)
        
        # 按优先级排序
        mentors.sort(key=lambda x: x.priority)
        
        logger.info(f"✅ 成功获取 {len(mentors)} 个启用的导师信息")
        return mentors
        
    except Exception as e:
        logger.error(f"❌ 获取启用的导师信息失败: {e}")
        raise HTTPException(status_code=500, detail=f"获取启用的导师信息失败: {str(e)}")

@router.get("/{agent_id}", response_model=MentorInfo)
async def get_mentor_by_id(agent_id: str):
    """
    根据ID获取导师信息
    
    Args:
        agent_id: 导师ID
        
    Returns:
        导师信息
    """
    try:
        logger.info(f"📋 获取导师信息: {agent_id}")
        
        agent = agent_manager.get_agent(agent_id)
        if not agent:
            raise HTTPException(status_code=404, detail=f"导师 {agent_id} 不存在")
        
        agent_config = agent_manager.agent_configs.get(agent_id, {})
        agent_info = agent.get_agent_info()
        
        mentor_info = MentorInfo(
            agent_id=agent_id,
            name=agent_config.get('name', agent_info.get('name', agent_id)),
            title=agent_info.get('title', ''),
            description=agent_config.get('description', ''),
            voice=agent_config.get('voice', agent_info.get('voice', 'Cherry')),
            expertise=agent_info.get('expertise', []),
            personality_traits=agent_info.get('personality_traits', []),
            investment_style=agent_info.get('investment_style', ''),
            famous_quotes=agent_info.get('famous_quotes', []),
            color=MENTOR_COLORS.get(agent_id, '#6B7280'),
            avatar=MENTOR_AVATARS.get(agent_id, f'https://api.dicebear.com/7.x/adventurer/svg?seed={agent_id}'),
            enabled=agent_config.get('enabled', True),
            priority=agent_config.get('priority', 1),
            registered_at=agent_config.get('registered_at', datetime.now().isoformat())
        )
        
        logger.info(f"✅ 成功获取导师信息: {agent_id}")
        return mentor_info
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"❌ 获取导师信息失败: {agent_id}, error: {e}")
        raise HTTPException(status_code=500, detail=f"获取导师信息失败: {str(e)}")

@router.get("/{agent_id}/config", response_model=MentorConfig)
async def get_mentor_config(agent_id: str):
    """
    获取导师配置信息
    
    Args:
        agent_id: 导师ID
        
    Returns:
        导师配置信息
    """
    try:
        logger.info(f"📋 获取导师配置: {agent_id}")
        
        agent_config = agent_manager.agent_configs.get(agent_id)
        if not agent_config:
            raise HTTPException(status_code=404, detail=f"导师 {agent_id} 配置不存在")
        
        config = MentorConfig(
            agent_id=agent_id,
            name=agent_config.get('name', agent_id),
            description=agent_config.get('description', ''),
            priority=agent_config.get('priority', 1),
            enabled=agent_config.get('enabled', True),
            voice=agent_config.get('voice', 'Cherry')
        )
        
        logger.info(f"✅ 成功获取导师配置: {agent_id}")
        return config
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"❌ 获取导师配置失败: {agent_id}, error: {e}")
        raise HTTPException(status_code=500, detail=f"获取导师配置失败: {str(e)}")

@router.put("/{agent_id}/config")
async def update_mentor_config(agent_id: str, update_request: MentorUpdateRequest):
    """
    更新导师配置
    
    Args:
        agent_id: 导师ID
        update_request: 更新请求
        
    Returns:
        更新结果
    """
    try:
        logger.info(f"📋 更新导师配置: {agent_id}")
        
        agent_config = agent_manager.agent_configs.get(agent_id)
        if not agent_config:
            raise HTTPException(status_code=404, detail=f"导师 {agent_id} 配置不存在")
        
        # 更新配置
        if update_request.name is not None:
            agent_config['name'] = update_request.name
        if update_request.description is not None:
            agent_config['description'] = update_request.description
        if update_request.priority is not None:
            agent_config['priority'] = update_request.priority
        if update_request.enabled is not None:
            agent_config['enabled'] = update_request.enabled
        if update_request.voice is not None:
            agent_config['voice'] = update_request.voice
        
        # 更新注册时间
        agent_config['updated_at'] = datetime.now().isoformat()
        
        logger.info(f"✅ 成功更新导师配置: {agent_id}")
        return {
            "message": "导师配置更新成功",
            "agent_id": agent_id,
            "updated_at": agent_config['updated_at']
        }
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"❌ 更新导师配置失败: {agent_id}, error: {e}")
        raise HTTPException(status_code=500, detail=f"更新导师配置失败: {str(e)}")

@router.post("/{agent_id}/enable")
async def enable_mentor(agent_id: str):
    """
    启用导师
    
    Args:
        agent_id: 导师ID
        
    Returns:
        启用结果
    """
    try:
        logger.info(f"📋 启用导师: {agent_id}")
        
        agent_config = agent_manager.agent_configs.get(agent_id)
        if not agent_config:
            raise HTTPException(status_code=404, detail=f"导师 {agent_id} 不存在")
        
        agent_config['enabled'] = True
        agent_config['updated_at'] = datetime.now().isoformat()
        
        logger.info(f"✅ 成功启用导师: {agent_id}")
        return {
            "message": "导师启用成功",
            "agent_id": agent_id,
            "enabled": True
        }
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"❌ 启用导师失败: {agent_id}, error: {e}")
        raise HTTPException(status_code=500, detail=f"启用导师失败: {str(e)}")

@router.post("/{agent_id}/disable")
async def disable_mentor(agent_id: str):
    """
    禁用导师
    
    Args:
        agent_id: 导师ID
        
    Returns:
        禁用结果
    """
    try:
        logger.info(f"📋 禁用导师: {agent_id}")
        
        agent_config = agent_manager.agent_configs.get(agent_id)
        if not agent_config:
            raise HTTPException(status_code=404, detail=f"导师 {agent_id} 不存在")
        
        agent_config['enabled'] = False
        agent_config['updated_at'] = datetime.now().isoformat()
        
        logger.info(f"✅ 成功禁用导师: {agent_id}")
        return {
            "message": "导师禁用成功",
            "agent_id": agent_id,
            "enabled": False
        }
        
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"❌ 禁用导师失败: {agent_id}, error: {e}")
        raise HTTPException(status_code=500, detail=f"禁用导师失败: {str(e)}")

@router.get("/stats/summary")
async def get_mentors_summary():
    """
    获取导师统计摘要
    
    Returns:
        导师统计信息
    """
    try:
        logger.info("📊 获取导师统计摘要")
        
        total_mentors = len(agent_manager.agents)
        enabled_mentors = len(agent_manager.get_enabled_agents())
        disabled_mentors = total_mentors - enabled_mentors
        
        # 按投资风格统计
        style_stats = {}
        for agent_id, agent in agent_manager.agents.items():
            agent_info = agent.get_agent_info()
            style = agent_info.get('investment_style', '未知')
            style_stats[style] = style_stats.get(style, 0) + 1
        
        # 按优先级统计
        priority_stats = {}
        for agent_id, config in agent_manager.agent_configs.items():
            priority = config.get('priority', 1)
            priority_stats[priority] = priority_stats.get(priority, 0) + 1
        
        summary = {
            "total_mentors": total_mentors,
            "enabled_mentors": enabled_mentors,
            "disabled_mentors": disabled_mentors,
            "enabled_ratio": enabled_mentors / total_mentors if total_mentors > 0 else 0,
            "style_distribution": style_stats,
            "priority_distribution": priority_stats,
            "last_updated": datetime.now().isoformat()
        }
        
        logger.info(f"✅ 成功获取导师统计摘要: 总数={total_mentors}, 启用={enabled_mentors}")
        return summary
        
    except Exception as e:
        logger.error(f"❌ 获取导师统计摘要失败: {e}")
        raise HTTPException(status_code=500, detail=f"获取导师统计摘要失败: {str(e)}")

@router.get("/templates/expertise")
async def get_expertise_templates():
    """
    获取专业领域模板
    
    Returns:
        专业领域模板列表
    """
    try:
        logger.info("📋 获取专业领域模板")
        
        # 从所有导师中收集专业领域
        all_expertise = set()
        for agent in agent_manager.agents.values():
            agent_info = agent.get_agent_info()
            expertise = agent_info.get('expertise', [])
            all_expertise.update(expertise)
        
        templates = {
            "available_expertise": sorted(list(all_expertise)),
            "popular_expertise": [
                "价值投资", "宏观投资", "多元思维", "宏观经济分析",
                "技术分析", "基本面分析", "风险管理", "资产配置"
            ],
            "expertise_categories": {
                "投资策略": ["价值投资", "成长投资", "指数投资", "量化投资"],
                "分析方法": ["技术分析", "基本面分析", "宏观分析", "行为分析"],
                "风险管理": ["风险管理", "资产配置", "投资组合", "对冲策略"],
                "专业领域": ["宏观经济", "国际贸易", "货币政策", "金融危机"]
            }
        }
        
        logger.info(f"✅ 成功获取专业领域模板: {len(templates['available_expertise'])} 个领域")
        return templates
        
    except Exception as e:
        logger.error(f"❌ 获取专业领域模板失败: {e}")
        raise HTTPException(status_code=500, detail=f"获取专业领域模板失败: {str(e)}")

@router.get("/templates/personality")
async def get_personality_templates():
    """
    获取性格特征模板
    
    Returns:
        性格特征模板列表
    """
    try:
        logger.info("📋 获取性格特征模板")
        
        # 从所有导师中收集性格特征
        all_traits = set()
        for agent in agent_manager.agents.values():
            agent_info = agent.get_agent_info()
            traits = agent_info.get('personality_traits', [])
            all_traits.update(traits)
        
        templates = {
            "available_traits": sorted(list(all_traits)),
            "personality_categories": {
                "投资风格": ["谨慎", "激进", "理性", "直觉", "耐心", "敏锐"],
                "沟通方式": ["幽默", "严肃", "直率", "委婉", "开放", "保守"],
                "思维方式": ["批判性", "创新性", "系统性", "发散性", "逻辑性", "创造性"],
                "工作态度": ["严谨", "灵活", "专注", "全面", "务实", "理想主义"]
            }
        }
        
        logger.info(f"✅ 成功获取性格特征模板: {len(templates['available_traits'])} 个特征")
        return templates
        
    except Exception as e:
        logger.error(f"❌ 获取性格特征模板失败: {e}")
        raise HTTPException(status_code=500, detail=f"获取性格特征模板失败: {str(e)}")

@router.post("/dynamic/generate", response_model=DynamicMentorResponse)
async def generate_dynamic_mentors(request: DynamicMentorRequest):
    """
    根据议题生成动态导师
    
    Args:
        request: 动态导师生成请求
        
    Returns:
        生成的导师信息
    """
    try:
        logger.info(f"🎯 收到动态导师生成请求: 议题='{request.topic}', 会话='{request.session_id}'")
        
        # 生成动态导师
        mentors = await agent_manager.generate_dynamic_mentors(request.topic, request.session_id)
        
        # 转换为MentorInfo格式
        mentor_infos = []
        for mentor in mentors:
            mentor_info = MentorInfo(
                agent_id=mentor['agent_id'],
                name=mentor['name'],
                title=mentor.get('title', ''),
                description=mentor.get('description', ''),
                voice=mentor.get('voice', 'Cherry'),
                expertise=mentor.get('expertise', []),
                personality_traits=mentor.get('personality_traits', []),
                investment_style=mentor.get('investment_style', ''),
                famous_quotes=mentor.get('famous_quotes', []),
                color=get_mentor_color(mentor['agent_id']),
                avatar=get_mentor_avatar(mentor['agent_id']),
                enabled=True,
                priority=2,
                registered_at=mentor.get('created_at', datetime.now().isoformat())
            )
            mentor_infos.append(mentor_info)
        
        response = DynamicMentorResponse(
            mentors=mentor_infos,
            topic=request.topic,
            session_id=request.session_id,
            generated_at=datetime.now().isoformat()
        )
        
        logger.info(f"✅ 成功生成 {len(mentor_infos)} 位动态导师")
        return response
        
    except Exception as e:
        logger.error(f"❌ 生成动态导师失败: {e}")
        raise HTTPException(status_code=500, detail=f"生成动态导师失败: {str(e)}")

@router.get("/dynamic/{session_id}")
async def get_session_dynamic_mentors(session_id: str):
    """
    获取会话的动态导师
    
    Args:
        session_id: 会话ID
        
    Returns:
        动态导师信息列表
    """
    try:
        logger.info(f"📋 获取会话 {session_id} 的动态导师")
        
        mentors = agent_manager.get_session_dynamic_mentors(session_id)
        topic = agent_manager.get_session_topic(session_id)
        
        return {
            "mentors": mentors,
            "topic": topic,
            "session_id": session_id
        }
        
    except Exception as e:
        logger.error(f"❌ 获取会话动态导师失败: {e}")
        raise HTTPException(status_code=500, detail="获取会话动态导师失败")

@router.delete("/dynamic/{session_id}")
async def cleanup_session_mentors(session_id: str):
    """
    清理会话的动态导师
    
    Args:
        session_id: 会话ID
    """
    try:
        logger.info(f"🗑️ 清理会话 {session_id} 的动态导师")
        
        agent_manager.cleanup_dynamic_mentors(session_id)
        
        return {"message": "动态导师清理成功"}
        
    except Exception as e:
        logger.error(f"❌ 清理动态导师失败: {e}")
        raise HTTPException(status_code=500, detail="清理动态导师失败")

File: frontend/src/config/mentors.ts
/**
 * 默认投资导师配置
 */
import { Mentor } from '../types/mentor'

// 投资风格颜色主题
export const MENTOR_COLORS = {
  blue: '#3B82F6',      // 蓝色 - 价值投资
  green: '#10B981',     // 绿色 - 宏观投资  
  purple: '#8B5CF6',    // 紫色 - 多元思维
  orange: '#F59E0B',    // 橙色 - 行为经济学
  red: '#EF4444',       // 红色 - 激进投资
  indigo: '#6366F1',    // 靛蓝 - 量化投资
  pink: '#EC4899',      // 粉色 - 创新投资
  teal: '#14B8A6',      // 青色 - 可持续投资
}

// 头像生成配置
export const AVATAR_STYLES = [
  'adventurer',
  'avataaars', 
  'big-ears',
  'big-smile',
  'croodles',
  'fun-emoji',
  'icons',
  'identicon',
  'initials',
  'lorelei',
  'micah',
  'miniavs',
  'open-peeps',
  'personas',
  'pixel-art'
]

// 生成随机头像URL
export function generateAvatarUrl(seed: string, style?: string): string {
  const selectedStyle = style || AVATAR_STYLES[Math.floor(Math.random() * AVATAR_STYLES.length)]
  return `https://api.dicebear.com/7.x/${selectedStyle}/svg?seed=${encodeURIComponent(seed)}`
}

// 默认投资导师配置
export const DEFAULT_MENTORS: Mentor[] = [
  {
    id: 'buffett',
    name: '沃伦·巴菲特',
    title: '伯克希尔·哈撒韦CEO',
    description: '价值投资之父，被誉为"奥马哈的先知"',
    expertise: ['价值投资', '长期持有', '企业分析', '护城河理论'],
    avatar: generateAvatarUrl('warren-buffett', 'adventurer'),
    color: MENTOR_COLORS.blue,
    personalityTraits: ['谦逊', '耐心', '理性', '幽默'],
    investmentStyle: '价值投资',
    famousQuotes: [
      '当别人恐惧时我贪婪，当别人贪婪时我恐惧',
      '时间是优秀企业的朋友，是平庸企业的敌人',
      '价格是你付出的，价值是你得到的'
    ],
    background: '从小展现出商业天赋，建立了投资界最成功的记录之一。通过伯克希尔·哈撒韦公司管理超过7000亿美元资产。'
  },
  {
    id: 'soros',
    name: '乔治·索罗斯',
    title: '量子基金创始人',
    description: '金融大鳄，以宏观经济投资和市场时机把握著称',
    expertise: ['宏观投资', '货币政策', '反身性理论', '市场时机'],
    avatar: generateAvatarUrl('george-soros', 'avataaars'),
    color: MENTOR_COLORS.green,
    personalityTraits: ['敏锐', '果断', '思辨', '前瞻'],
    investmentStyle: '宏观投资',
    famousQuotes: [
      '金融市场天生就不稳定',
      '重要的不是你是否正确，而是你正确时赚了多少钱，错误时亏了多少钱',
      '市场参与者的认知天生就是有缺陷的'
    ],
    background: '哲学出身的投资大师，以"反身性理论"闻名。1992年成功做空英镑，被称为"击败英格兰银行的人"。'
  },
  {
    id: 'munger',
    name: '查理·芒格',
    title: '伯克希尔·哈撒韦副主席',
    description: '多元思维模型倡导者，巴菲特的黄金搭档',
    expertise: ['多元思维', '跨学科分析', '理性决策', '心理学'],
    avatar: generateAvatarUrl('charlie-munger', 'big-ears'),
    color: MENTOR_COLORS.purple,
    personalityTraits: ['睿智', '直率', '博学', '独立'],
    investmentStyle: '多元思维投资',
    famousQuotes: [
      '获得智慧是一种道德责任',
      '如果你想要说服别人，要诉诸利益，而非理性',
      '学会学习如何正确地思考，这是最重要的技能'
    ],
    background: '律师出身，以跨学科思维著称。与巴菲特合作超过50年，被称为"行走的书库"。'
  },
  {
    id: 'krugman',
    name: '保罗·克鲁格曼',
    title: '诺贝尔经济学奖得主',
    description: '新贸易理论和新经济地理学创始人，经济政策分析专家',
    expertise: ['宏观经济学', '国际贸易', '货币政策', '经济周期'],
    avatar: generateAvatarUrl('paul-krugman', 'croodles'),
    color: MENTOR_COLORS.orange,
    personalityTraits: ['学术严谨', '批判思维', '数据驱动', '政策敏感'],
    investmentStyle: '宏观经济分析',
    famousQuotes: [
      '经济学不是一门精确科学，但它仍然是一门科学',
      '市场可能在很长时间内保持非理性',
      '政策制定者需要基于证据而非意识形态做决策'
    ],
    background: 'MIT教授，2008年诺贝尔经济学奖得主。以其对国际贸易和经济地理学的贡献而闻名，同时也是著名的经济政策评论家。'
  }
]

// 导师模板（用于创建自定义导师的参考）
export const MENTOR_TEMPLATES = {
  expertise: [
    '价值投资', '成长投资', '指数投资', '量化投资', '技术分析',
    '基本面分析', '宏观投资', '微观分析', '风险管理', '资产配置',
    '行为金融学', '市场心理学', '企业分析', '财务分析', '估值方法',
    '衍生品', '固定收益', '另类投资', 'ESG投资', '科技投资'
  ],
  personalityTraits: [
    '谨慎', '激进', '理性', '直觉', '耐心', '敏锐', '创新', '保守',
    '幽默', '严肃', '乐观', '悲观', '务实', '理想主义', '批判性', '开放性'
  ],
  investmentStyles: [
    '价值投资', '成长投资', '动量投资', '逆向投资', '指数投资',
    '量化投资', '宏观投资', '事件驱动', '套利投资', '长期投资',
    '短期交易', '技术分析', '基本面分析', '多元化投资', '集中投资'
  ]
}

// 获取可用的颜色（排除已使用的）
export function getAvailableColors(usedColors: string[]): string[] {
  const allColors = Object.values(MENTOR_COLORS)
  return allColors.filter(color => !usedColors.includes(color))
}

// 为新导师分配颜色
export function assignMentorColor(existingMentors: Mentor[]): string {
  const usedColors = existingMentors.map(mentor => mentor.color)
  const availableColors = getAvailableColors(usedColors)
  
  if (availableColors.length > 0) {
    return availableColors[0]
  }
  
  // 如果所有颜色都被使用，随机选择一个
  const allColors = Object.values(MENTOR_COLORS)
  return allColors[Math.floor(Math.random() * allColors.length)]
}
